### ISSUE TITLE
wp-config.php parsing issue with commented out define()

### ISSUE THREAD

**Issue by ChrisDean** [wp-cli/wp-cli#2019]
Hi guys

Have been using wp-cli for a while now and was getting some weird silent failures on a multisite install which is being cloned in its entirety to form various levels of test environment.

Have just upgraded to 0.20.0 and the issue isn't silent anymore (thanks for that :)) and I can now see what the issue is and thought I'd flag it here in case it's of use to someone else/something that can be rectified in a future version.

Basically while setting up I'm being a little lazy and have the following in my config file

    define('DOMAIN_CURRENT_SITE', 'exa.example.com');
    #define('DOMAIN_CURRENT_SITE', 'bta.example.com');
    #define('DOMAIN_CURRENT_SITE', 'www.example.com');
    #define('DOMAIN_CURRENT_SITE', 'uat.example.com');

It looks as if (without specifying the --url parameter) wp-cli is parsing the wp-config file and using the last value for  DOMAIN_CURRENT_SITE regardless of it being commented out.

So when I'm trying to run wp-cli against exa.example.com it think's I'm trying to run against uat.example.com and failing because uat.example.com is not a defined site in the  multisite install in this copy of the db.

Make sense?

If not would be happy to clarify but I think generally the parsing of the wp-config needs to be cleaned up a little in wp-cli - If I get a chance I'll do it myself and supply it back but right now am crazy busy :)

At least maybe the bug report will help!

Cheers and thanks for a great tool!

Chris

**Comment by danielbachhuber** [comment]
> Make sense?

Yep. Looks like it's a bug with this regex: https://github.com/wp-cli/wp-cli/blob/ad54db39f461d5d00cbb8554178dc90d72fdce1f/php/WP_CLI/Runner.php#L214

> If not would be happy to clarify but I think generally the parsing of the wp-config needs to be cleaned up a little in wp-cli

Known issue #1631

tl;dr; we shouldn't be parsing wp-config.php with regex.
